WIP! Volto Light Theme Training docs. (#861)
plone/training: docs/volto-light-theme/creating-new-project.md

---
myst:
  html_meta:
    'description': 'Learn How to customize the Footer of the page'
    'property=og:description': 'Learn How to customize the Footer of the page'
    'property=og:title': 'Footer customization'
    'keywords': 'Plone, Volto, Training, Theme, Footer'
---

> # Creating New Project

Reviewer: ```suggestion
# Creating a new project
```

```
pipx run cookieplone
```

It is going to ask you multiple question. If you happy with default keep pressing enter. Otherwise overwrite this according to your needs.

```
 [1/17] Project Title (Project Title): my-vlt-project
  [2/17] Project Description (A new project using Plone 6.): My Project
  [3/17] Project Slug (Used for repository id) (my-vlt-project): my-project
  [4/17] Project URL (without protocol) (my-project.example.com): my-project.example.com
  [5/17] Author (Plone Foundation): <your name>
  [6/17] Author E-mail (collective@plone.org):
  [7/17] Should we use prerelease versions? (No):
  [8/17] Plone Version (6.0.13):
  [9/17] Volto Version (18.0.0):
  [10/17] Python Package Name (my.project):
  [11/17] Volto Addon Name (volto-my-project):
```

After creating project you have to build the volto project.

Run
```
make install
``

This will install all your backend as well as frontend project.

For running backend you have to run

```
make backend-start

```

For running frontend

```
make frontend-start

```

Please commit your inital commit by running

```
git add .
git commit -m 'initial-commit' --no-verify
```

> Walk throught the boiler plate and explain them.

Reviewer: I would replace all the foregoing content with a link to the new documentation. You can use Intersphinx and MyST syntax.

Of course, before you can install, you need to satisfy system requirements.

```suggestion
{doc}`plone:docs/install/create-project-cookieplone`
```

    'keywords': 'Plone, Volto, Training, Theme, Footer'
---

# Creating a New Project with Plone and Volto

Follow the steps below to create a new project using Plone 6 and Volto. This guide assumes you have `pipx` installed.

### Step 1: Install CookiePlone

First, install `cookieplone`, a tool for setting up new Plone projects, using `pipx`:

```bash
pipx install cookieplone
```

### Step 2: Create the Project

> Once `cookieplone` is installed, run the command below to start the project creation process:

Author: done

```bash
pipx run cookieplone
```

> You’ll be prompted with several questions. You can accept the default values by pressing **Enter** or overwrite them according to your preferences. Here are the prompts you’ll encounter:

Author: done

> ```

Reviewer: This lexer is `console`.

````suggestion
```console
````

Author: done

[1/17] Project Title (Project Title): my-vlt-project
[2/17] Project Description (A new project using Plone 6.): My Project
[3/17] Project Slug (Used for repository id) (my-vlt-project): my-project
[4/17] Project URL (without protocol) (my-project.example.com): my-project.example.com
[5/17] Author (Plone Foundation): <your name>
[6/17] Author E-mail (collective@plone.org):
[7/17] Should we use prerelease versions? (No):
[8/17] Plone Version (6.0.13):
[9/17] Volto Version (18.0.0):
[10/17] Python Package Name (my.project):
[11/17] Volto Addon Name (volto-my-project):
```

- **Project Title**: The title of your project (e.g., `my-vlt-project`).
- **Project Description**: A short description of your project.
- **Project Slug**: A repository-friendly identifier for your project (e.g., `my-project`).
- **Project URL**: The URL where your project will be hosted (e.g., `my-project.example.com`).
- **Author**: Your name.
- **Author E-mail**: Your email address (can be left blank).
- **Plone Version**: The Plone version you’re using (e.g., `6.0.13`).
- **Volto Version**: The Volto version to be used (e.g., `18.0.0`).
- **Python Package Name**: The Python package name (e.g., `my.project`).

> - **Volto Addon Name**: The Volto add-on name (e.g., `volto-my-project`).

Reviewer: The above is duplicitous of the console output. I would remove it.

Author: done

After answering the prompts, the project will be created.

### Step 3: Install Dependencies

> Once the project is created, you need to build both the backend and frontend.

Reviewer: Install is a one-time thing. "build" is usually associated with compiling static assets or making a release and happens many times.

```suggestion
Once the project is created, you need to install dependencies for both the backend and frontend.
```

And looking at this process again, there are multiple issues. I won't spend any more time correcting stuff that already exists. It's not a good use of my time.

I think you would serve participants better by linking to the documentation at https://6.docs.plone.org/install/create-project-cookieplone.html#install-the-project, or instead of rewriting stuff with mistakes, you can copy-paste from the source of documentation.

Author: done
